### DISCUSSION TITLE
Eglot cannot start rust-analyzer in a project with `rust-toolchain` file

### DISCUSSION THREAD

**Discussion by tbro** [joaotavora/eglot#1271] [General]
I compiled emacs today to try 29.1 with native eglot. Everything went well. I downloaded a new open-source project and tried to open a file and rust-analyzer crashed right away. Looking at stderr I saw ```error: 'rust-analyzer' is not installed for the toolchain '1.63.0-x86_64-unknown-linux-gnu'```. I checked `exec-path` and a few other possibilities. But once I remove `rust-toolchain`, in the root of the project, I closed and reopened the buffer, and I could see eglot indexing the project. contents of the `rust-toolchain` file were

    [toolchain]
    channel = "1.63.0"
    components = [ "rustc", "clippy", "rustfmt" ]

I'm not very knowledgeable about inter-working of eglot with the different modes, so if I'm reporting in the wrong place sorry for the noise

**Comment by mohkale** [comment]
This is not a eglot error, rust-analyzer is refusing to start. Try running it directly in your project and if it still fails then you'll have to figure out why. I'm guessing you need to install rust-analyzer in your toolchain file? Although I'm not that familiar with rust.

**Reply by tbro** [reply]
Thank you for your response. Yes. It turns out I had installed some sort of shim in ~/.local/bin which pointed to the rust-analyzer installed through `rustup component add`. Since there was no such component back in toolchain 1.63, rust-analyzer failed to load. Reinstalling rust-anlyzer using the latest release binary to ~/.local/bin fixed my issue.
